### DISCUSSION TITLE
Unique class instance for every terminal.

### DISCUSSION THREAD

**Discussion by iscapla** [daniele77/cli#189] [Q&A]
Hello,

Imagine the scenario when every terminal needs its own class instance to interact and other sessions are not allowed to modified or even access others.

One menu command could be something like this:
```cpp
rootMenu->Insert(
                "free_function",
                [](std::ostream& out){
                              auto myID = get_unique_terminal_id();
                              auto my_instance = dict[myID];
                              my_instance.foo();
                },
                "Call foo function on my private space" );
```

As you can see, we only need a function or a variable that returns the terminal unique id.

Is this possible?
Does telnet server provide this kind of feature?

Thanks.

**Comment by daniele77** [comment]
Hi @iscapla ,
thank you for your suggestion.
Unfortunately, the library does not provide this feature, yet.
However, it would be a nice feature to have: it should be added to the development plan.
Please check issue #49 and tell us if it's the same request.

**Reply by iscapla** [reply]
Hi @daniele77 ,
Yes, I think so.
To be able to interact with a concrete terminal you also need something to identify it. Maybe a general handler that maintains the unique identifier (something similar to how sockets or files work), and other stuff directly related to this terminal.

**Reply by daniele77** [reply]
Ideally, you should pass a "session", not a terminal.
